[Yarn-test] Cluster Error: jep.JepException: <class 'ImportError'>

# [fashino_mnist.py](https://github.com/intel-analytics/BigDL/tree/branch-2.0/python/orca/example/learn/pytorch/fashion_mnist)
### cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise
## Yarn-Cluster Exception
```
creating: createZooKerasAccuracy
creating: createEstimator
creating: createEveryEpoch
creating: createMaxEpoch
2021-11-27 05:17:21 INFO  DistriOptimizer$:824 - caching training rdd ...
2021-11-27 05:17:50 INFO  DistriOptimizer$:650 - Cache thread models...
2021-11-27 05:17:51 INFO  DistriOptimizer$:652 - Cache thread models... done
2021-11-27 05:17:51 INFO  DistriOptimizer$:162 - Count dataset
2021-11-27 05:18:04 INFO  DistriOptimizer$:166 - Count dataset complete. Time elapsed: 12.86615251s
2021-11-27 05:18:16 WARN  DistriOptimizer$:168 - If the dataset is built directly from RDD[Minibatch], the data in each minibatch is fixed, and a single minibatch is randomly selected in each partition. If the dataset is transformed from RDD[Sample], each minibatch will be constructed on the fly from random samples, which is better for convergence.
2021-11-27 05:18:16 INFO  DistriOptimizer$:174 - config  {
	computeThresholdbatchSize: 100
	maxDropPercentage: 0.0
	warmupIterationNum: 200
	isLayerwiseScaled: false
	dropPercentage: 0.0
 }
2021-11-27 05:18:16 INFO  DistriOptimizer$:178 - Shuffle data
2021-11-27 05:18:16 INFO  DistriOptimizer$:181 - Shuffle data complete. Takes 2.89354E-4s
2021-11-27 05:18:17 ERROR DistriOptimizer$:935 - Error: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intel.analytics.bigdl.dllib.keras.layers.utils.KerasUtils$.invokeMethod(KerasUtils.scala:302)
	at com.intel.analytics.bigdl.dllib.keras.layers.utils.KerasUtils$.invokeMethodWithEv(KerasUtils.scala:329)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalOptimizerUtil$.optimizeModels(Topology.scala:710)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:910)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:1123)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:793)
	at com.intel.analytics.bigdl.dllib.estimator.Estimator.train(Estimator.scala:190)
	at com.intel.analytics.bigdl.dllib.estimator.python.PythonEstimator.estimatorTrainMiniBatch(PythonEstimator.scala:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
Caused by: jep.JepException: jep.JepException: <class 'ImportError'>: /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/lib-dynload/
_posixsubprocess.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$.threadExecute(PythonInterpreter.scala:98)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$.exec(PythonInterpreter.scala:108)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.optimType$lzycompute(TorchOptim.scala:55)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.optimType(TorchOptim.scala:38)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.updateHyperParameter(TorchOptim.scala:166)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$$anonfun$optimize$5.apply(DistriOptimizer.scala:420)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$$anonfun$optimize$5.apply(DistriOptimizer.scala:419)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:116)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$.optimize(DistriOptimizer.scala:419)
	... 23 more
Caused by: jep.JepException: <class 'ImportError'>: /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/lib-dynload/_posixsubprocess.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/subprocess.<module>(subprocess.py:152)
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/platform.<module>(platform.py:116)
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/torch/__init__.<module>(__init__.py:14)
	at <string>.<module>(<string>:2)
	at jep.Jep.exec(Native Method)
	at jep.Jep.exec(Jep.java:478)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:106)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:105)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:105)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more

Traceback (most recent call last):
  File "fashion_mnist.py", line 196, in <module>
    main()
  File "fashion_mnist.py", line 170, in main
    checkpoint_trigger=EveryEpoch())
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/bigdl/orca/learn/pytorch/pytorch_spark_estimator.py", line 170, in fit
    checkpoint_trigger, val_fset, self.metrics)
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/bigdl/dllib/estimator/estimator.py", line 167, in train_minibatch
    validation_method)
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/bigdl/dllib/utils/file_utils.py", line 164, in callZooFunc
    raise e
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/bigdl/dllib/utils/file_utils.py", line 158, in callZooFunc
    java_result = api(*args)
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/py4j-0.10.7-src.zip/py4j/java_gateway.py", line 1257, in __call__
  File "/dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/py4j-0.10.7-src.zip/py4j/protocol.py", line 328, in get_return_value
py4j.protocol.Py4JJavaError: An error occurred while calling o76.estimatorTrainMiniBatch.
: jep.JepException: jep.JepException: <class 'ImportError'>: /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/lib-dynload/_posixsubprocess.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$.threadExecute(PythonInterpreter.scala:98)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$.exec(PythonInterpreter.scala:108)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.optimType$lzycompute(TorchOptim.scala:55)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.optimType(TorchOptim.scala:38)
	at com.intel.analytics.bigdl.orca.net.TorchOptim.updateHyperParameter(TorchOptim.scala:166)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$$anonfun$optimize$5.apply(DistriOptimizer.scala:420)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$$anonfun$optimize$5.apply(DistriOptimizer.scala:419)
	at scala.collection.immutable.Map$Map1.foreach(Map.scala:116)
	at com.intel.analytics.bigdl.dllib.optim.DistriOptimizer$.optimize(DistriOptimizer.scala:419)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intel.analytics.bigdl.dllib.keras.layers.utils.KerasUtils$.invokeMethod(KerasUtils.scala:302)
	at com.intel.analytics.bigdl.dllib.keras.layers.utils.KerasUtils$.invokeMethodWithEv(KerasUtils.scala:329)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalOptimizerUtil$.optimizeModels(Topology.scala:710)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:910)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:1123)
	at com.intel.analytics.bigdl.dllib.keras.models.InternalDistriOptimizer.train(Topology.scala:793)
	at com.intel.analytics.bigdl.dllib.estimator.Estimator.train(Estimator.scala:190)
	at com.intel.analytics.bigdl.dllib.estimator.python.PythonEstimator.estimatorTrainMiniBatch(PythonEstimator.scala:117)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)
Caused by: jep.JepException: <class 'ImportError'>: /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/lib-dynload/_posixsubprocess.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _Py_write_noraise
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/subprocess.<module>(subprocess.py:152)
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/platform.<module>(platform.py:116)
	at /dir2/yarn/nm_0/usercache/root/appcache/application_1635129125298_1545/container_1635129125298_1545_02_000001/python_env/lib/python3.7/site-packages/torch/__init__.<module>(__init__.py:14)
	at <string>.<module>(<string>:2)
	at jep.Jep.exec(Native Method)
	at jep.Jep.exec(Jep.java:478)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply$mcV$sp(PythonInterpreter.scala:106)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:105)
	at com.intel.analytics.bigdl.orca.utils.PythonInterpreter$$anonfun$1.apply(PythonInterpreter.scala:105)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more

Stopping orca context
```

--- a/python/dllib/examples/run-example-tests-yarn-integration.sh
+++ b/python/dllib/examples/run-example-tests-yarn-integration.sh
@@ -10,7 +10,6 @@ ls /tmp/mnist
 python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/lenet/lenet5.py --cluster-mode yarn-client -n 1
 exit_status=$?
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib lenet5 failed"
   exit $exit_status
 fi
@@ -34,7 +33,6 @@ python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/inception/inception.py 
         -i 20
 exit_status=$?
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib inception failed"
   exit $exit_status
 fi
@@ -52,7 +50,6 @@ ls /tmp/news20
 python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/textclassifier/textclassifier.py --on-yarn --max_epoch 3 --model cnn
 exit_status=$?
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib textclassifier failed"
   exit $exit_status
 fi
@@ -67,7 +64,6 @@ start=$(date "+%s")
 #run the example
 python ${BIGDL_ROOT}/python/dllib/examples/autograd/custom.py --cluster-mode "yarn-client"
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib autograd custom failed"
   exit $exit_status
 fi
@@ -82,7 +78,6 @@ start=$(date "+%s")
 #run the example
 python ${BIGDL_ROOT}/python/dllib/examples/autograd/customloss.py --cluster-mode "yarn-client"
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib autograd customloss failed"
   exit $exit_status
 fi
@@ -107,7 +102,6 @@ python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageInference/ImageInferenc
   -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
   -f ${HDFS_URI}/kaggle/train_100 --cluster-mode "yarn-client"
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib nnframes_imageInference failed"
   exit $exit_status
 fi
@@ -124,7 +118,6 @@ python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageTransferLearning/ImageT
   -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
   -f ${HDFS_URI}/dogs_cats/samples --nb_epoch 2 --cluster-mode "yarn-client"
 if [ $exit_status -ne 0 ]; then
-  clear_up
   echo "dllib nnframes_imageTransfer learning failed"
   exit $exit_status
 fi

--- a/python/dllib/examples/run-example-tests-yarn-integration.sh
+++ b/python/dllib/examples/run-example-tests-yarn-integration.sh
@@ -1,5 +1,19 @@
 #!/bin/bash
-echo "#1 start test for dllib lenet5"
+echo "#12 start test for pytorch cifar10"
+#timer  
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "orca pytorch cifar10 failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#12 Total time cost ${time} seconds"
+#echo "#1 start test for dllib lenet5"
 
 ##timer
 #start=$(date "+%s")

--- a/python/dllib/examples/run-example-tests-yarn-integration.sh
+++ b/python/dllib/examples/run-example-tests-yarn-integration.sh
@@ -1,18 +1,4 @@
 #!/bin/bash
-echo "#12 start test for pytorch cifar10"
-#timer  
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch cifar10 failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#12 Total time cost ${time} seconds"
 #echo "#1 start test for dllib lenet5"
 
 ##timer

--- a/python/dllib/examples/run-example-tests-yarn-integration.sh
+++ b/python/dllib/examples/run-example-tests-yarn-integration.sh
@@ -1,133 +1,133 @@
 #!/bin/bash
 echo "#1 start test for dllib lenet5"
 
-#timer
-start=$(date "+%s")
-#run the example
-rm -rf /tmp/mnist
-${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/mnist /tmp/mnist
-ls /tmp/mnist
-python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/lenet/lenet5.py --cluster-mode yarn-client -n 1
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib lenet5 failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#1 Total time cost ${time} seconds"
-
-echo "#2 start test for dllib inception"
-
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/inception/inception.py -f ${HDFS_URI}/imagenet-mini \
-	--batchSize 128 \
-	--learningRate 0.065 \
-	--weightDecay 0.0002 \
-	--executor-memory 20g \
-	--driver-memory 20g \
-	--executor-cores 4 \
-	--num-executors 4 \
-        -i 20
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib inception failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#2 Total time cost ${time} seconds"
-
-echo "#3 start test for dllib textclassifier"
-#timer
-start=$(date "+%s")
-#run the example
-rm -rf /tmp/news20
-${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/news20 /tmp/news20
-ls /tmp/news20
-python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/textclassifier/textclassifier.py --on-yarn --max_epoch 3 --model cnn
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib textclassifier failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#3 Total time cost ${time} seconds"
-
-
-echo "#4 start test for dllib autograd custom"
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/dllib/examples/autograd/custom.py --cluster-mode "yarn-client"
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib autograd custom failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#4 Total time cost ${time} seconds"
-
-
-echo "#5 start test for dllib autograd customloss"
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/dllib/examples/autograd/customloss.py --cluster-mode "yarn-client"
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib autograd customloss failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#5 Total time cost ${time} seconds"
-
-
-echo "#6 start test for dllib nnframes_imageInference"
-
-if [ -f analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model ]; then
-  echo "analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model already exists."
-else
-  wget $FTP_URI/analytics-zoo-models/image-classification/bigdl_inception-v1_imagenet_0.4.0.model \
-    -P analytics-zoo-models
-fi
-
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageInference/ImageInferenceExample.py \
-  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
-  -f ${HDFS_URI}/kaggle/train_100 --cluster-mode "yarn-client"
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib nnframes_imageInference failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#6 Total time cost ${time} seconds"
-
-
-echo "#7 start test for dllib nnframes_imageTransfer learning"
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageTransferLearning/ImageTransferLearningExample.py \
-  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
-  -f ${HDFS_URI}/dogs_cats/samples --nb_epoch 2 --cluster-mode "yarn-client"
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "dllib nnframes_imageTransfer learning failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#7 Total time cost ${time} seconds"
+##timer
+#start=$(date "+%s")
+##run the example
+#rm -rf /tmp/mnist
+#${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/mnist /tmp/mnist
+#ls /tmp/mnist
+#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/lenet/lenet5.py --cluster-mode yarn-client -n 1
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib lenet5 failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#1 Total time cost ${time} seconds"
+#
+#echo "#2 start test for dllib inception"
+#
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/inception/inception.py -f ${HDFS_URI}/imagenet-mini \
+#	--batchSize 128 \
+#	--learningRate 0.065 \
+#	--weightDecay 0.0002 \
+#	--executor-memory 20g \
+#	--driver-memory 20g \
+#	--executor-cores 4 \
+#	--num-executors 4 \
+#        -i 20
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib inception failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#2 Total time cost ${time} seconds"
+#
+#echo "#3 start test for dllib textclassifier"
+##timer
+#start=$(date "+%s")
+##run the example
+#rm -rf /tmp/news20
+#${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/news20 /tmp/news20
+#ls /tmp/news20
+#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/textclassifier/textclassifier.py --on-yarn --max_epoch 3 --model cnn
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib textclassifier failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#3 Total time cost ${time} seconds"
+#
+#
+#echo "#4 start test for dllib autograd custom"
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/dllib/examples/autograd/custom.py --cluster-mode "yarn-client"
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib autograd custom failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#4 Total time cost ${time} seconds"
+#
+#
+#echo "#5 start test for dllib autograd customloss"
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/dllib/examples/autograd/customloss.py --cluster-mode "yarn-client"
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib autograd customloss failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#5 Total time cost ${time} seconds"
+#
+#
+#echo "#6 start test for dllib nnframes_imageInference"
+#
+#if [ -f analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model ]; then
+#  echo "analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model already exists."
+#else
+#  wget $FTP_URI/analytics-zoo-models/image-classification/bigdl_inception-v1_imagenet_0.4.0.model \
+#    -P analytics-zoo-models
+#fi
+#
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageInference/ImageInferenceExample.py \
+#  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
+#  -f ${HDFS_URI}/kaggle/train_100 --cluster-mode "yarn-client"
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib nnframes_imageInference failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#6 Total time cost ${time} seconds"
+#
+#
+#echo "#7 start test for dllib nnframes_imageTransfer learning"
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageTransferLearning/ImageTransferLearningExample.py \
+#  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
+#  -f ${HDFS_URI}/dogs_cats/samples --nb_epoch 2 --cluster-mode "yarn-client"
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "dllib nnframes_imageTransfer learning failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#7 Total time cost ${time} seconds"

--- a/python/dllib/examples/run-example-tests-yarn-integration.sh
+++ b/python/dllib/examples/run-example-tests-yarn-integration.sh
@@ -1,133 +1,133 @@
 #!/bin/bash
-#echo "#1 start test for dllib lenet5"
+echo "#1 start test for dllib lenet5"
 
-##timer
-#start=$(date "+%s")
-##run the example
-#rm -rf /tmp/mnist
-#${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/mnist /tmp/mnist
-#ls /tmp/mnist
-#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/lenet/lenet5.py --cluster-mode yarn-client -n 1
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib lenet5 failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#1 Total time cost ${time} seconds"
-#
-#echo "#2 start test for dllib inception"
-#
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/inception/inception.py -f ${HDFS_URI}/imagenet-mini \
-#	--batchSize 128 \
-#	--learningRate 0.065 \
-#	--weightDecay 0.0002 \
-#	--executor-memory 20g \
-#	--driver-memory 20g \
-#	--executor-cores 4 \
-#	--num-executors 4 \
-#        -i 20
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib inception failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#2 Total time cost ${time} seconds"
-#
-#echo "#3 start test for dllib textclassifier"
-##timer
-#start=$(date "+%s")
-##run the example
-#rm -rf /tmp/news20
-#${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/news20 /tmp/news20
-#ls /tmp/news20
-#python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/textclassifier/textclassifier.py --on-yarn --max_epoch 3 --model cnn
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib textclassifier failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#3 Total time cost ${time} seconds"
-#
-#
-#echo "#4 start test for dllib autograd custom"
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/dllib/examples/autograd/custom.py --cluster-mode "yarn-client"
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib autograd custom failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#4 Total time cost ${time} seconds"
-#
-#
-#echo "#5 start test for dllib autograd customloss"
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/dllib/examples/autograd/customloss.py --cluster-mode "yarn-client"
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib autograd customloss failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#5 Total time cost ${time} seconds"
-#
-#
-#echo "#6 start test for dllib nnframes_imageInference"
-#
-#if [ -f analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model ]; then
-#  echo "analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model already exists."
-#else
-#  wget $FTP_URI/analytics-zoo-models/image-classification/bigdl_inception-v1_imagenet_0.4.0.model \
-#    -P analytics-zoo-models
-#fi
-#
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageInference/ImageInferenceExample.py \
-#  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
-#  -f ${HDFS_URI}/kaggle/train_100 --cluster-mode "yarn-client"
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib nnframes_imageInference failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#6 Total time cost ${time} seconds"
-#
-#
-#echo "#7 start test for dllib nnframes_imageTransfer learning"
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageTransferLearning/ImageTransferLearningExample.py \
-#  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
-#  -f ${HDFS_URI}/dogs_cats/samples --nb_epoch 2 --cluster-mode "yarn-client"
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "dllib nnframes_imageTransfer learning failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#7 Total time cost ${time} seconds"
+#timer
+start=$(date "+%s")
+#run the example
+rm -rf /tmp/mnist
+${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/mnist /tmp/mnist
+ls /tmp/mnist
+python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/lenet/lenet5.py --cluster-mode yarn-client -n 1
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib lenet5 failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#1 Total time cost ${time} seconds"
+
+echo "#2 start test for dllib inception"
+
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/inception/inception.py -f ${HDFS_URI}/imagenet-mini \
+	--batchSize 128 \
+	--learningRate 0.065 \
+	--weightDecay 0.0002 \
+	--executor-memory 20g \
+	--driver-memory 20g \
+	--executor-cores 4 \
+	--num-executors 4 \
+        -i 20
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib inception failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#2 Total time cost ${time} seconds"
+
+echo "#3 start test for dllib textclassifier"
+#timer
+start=$(date "+%s")
+#run the example
+rm -rf /tmp/news20
+${HADOOP_HOME}/bin/hadoop fs -get ${HDFS_URI}/news20 /tmp/news20
+ls /tmp/news20
+python ${BIGDL_ROOT}/python/dllib/src/bigdl/dllib/models/textclassifier/textclassifier.py --on-yarn --max_epoch 3 --model cnn
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib textclassifier failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#3 Total time cost ${time} seconds"
+
+
+echo "#4 start test for dllib autograd custom"
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/dllib/examples/autograd/custom.py --cluster-mode "yarn-client"
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib autograd custom failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#4 Total time cost ${time} seconds"
+
+
+echo "#5 start test for dllib autograd customloss"
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/dllib/examples/autograd/customloss.py --cluster-mode "yarn-client"
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib autograd customloss failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#5 Total time cost ${time} seconds"
+
+
+echo "#6 start test for dllib nnframes_imageInference"
+
+if [ -f analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model ]; then
+  echo "analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model already exists."
+else
+  wget $FTP_URI/analytics-zoo-models/image-classification/bigdl_inception-v1_imagenet_0.4.0.model \
+    -P analytics-zoo-models
+fi
+
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageInference/ImageInferenceExample.py \
+  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
+  -f ${HDFS_URI}/kaggle/train_100 --cluster-mode "yarn-client"
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib nnframes_imageInference failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#6 Total time cost ${time} seconds"
+
+
+echo "#7 start test for dllib nnframes_imageTransfer learning"
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/dllib/examples/nnframes/imageTransferLearning/ImageTransferLearningExample.py \
+  -m analytics-zoo-models/bigdl_inception-v1_imagenet_0.4.0.model \
+  -f ${HDFS_URI}/dogs_cats/samples --nb_epoch 2 --cluster-mode "yarn-client"
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "dllib nnframes_imageTransfer learning failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#7 Total time cost ${time} seconds"

--- a/python/dllib/src/bigdl/dllib/utils/spark.py
+++ b/python/dllib/src/bigdl/dllib/utils/spark.py
@@ -189,6 +189,7 @@ class SparkRunner:
                          "spark.executorEnv.PYTHONHOME": executor_python_env,
                          "spark.executor.extraLibraryPath": ld_path,
                          "spark.executorEnv.LD_PRELOAD": preload_so})
+            conf["spark.yarn.appMasterEnv.LD_PRELOAD"] = preload_so
             if spark_yarn_archive:
                 conf["spark.yarn.archive"] = spark_yarn_archive
             zoo_bigdl_path_on_executor = ":".join(

--- a/python/dllib/src/bigdl/dllib/utils/spark.py
+++ b/python/dllib/src/bigdl/dllib/utils/spark.py
@@ -155,6 +155,7 @@ class SparkRunner:
         pack_env = False
         assert penv_archive or conda_name, \
             "You should either specify penv_archive or conda_name explicitly"
+        return_value = 1
         try:
 
             if not penv_archive:
@@ -208,11 +209,11 @@ class SparkRunner:
             print(sys_args)
             submit_commnad = "spark-submit " + submit_args + " " + conf + " " + sys_args
             print(submit_commnad)
-            return os.system(submit_commnad)
+            return_value = os.system(submit_commnad)
         finally:
             if conda_name and penv_archive and pack_env:
                 os.remove(penv_archive)
-            return 1
+            return return_value
 
     def init_spark_standalone(self,
                               num_executors,

--- a/python/orca/example/learn/pytorch/cifar10/cifar10.py
+++ b/python/orca/example/learn/pytorch/cifar10/cifar10.py
@@ -49,8 +49,8 @@ parser.add_argument('--batch_size', type=int, default=64, help='The training bat
 parser.add_argument('--epochs', type=int, default=2, help='The number of epochs to train for')
 parser.add_argument('--data_dir', type=str, default="./data", help='The path to dataset')
 parser.add_argument('--download', type=bool, default=True, help='Download dataset or not')
-parser.add_argument("--executor_memory", type=str, default="2g", help="executor memory")
-parser.add_argument("--driver_memory", type=str, default="2g", help="driver memory")
+parser.add_argument("--executor_memory", type=str, default="5g", help="executor memory")
+parser.add_argument("--driver_memory", type=str, default="5g", help="driver memory")
 args = parser.parse_args()
 
 if args.cluster_mode == "local":

--- a/python/orca/example/run-example-tests-yarn-integration.sh
+++ b/python/orca/example/run-example-tests-yarn-integration.sh
@@ -1,217 +1,217 @@
-##!/bin/bash
-#echo "#11 start test for orca data spark_pandas"
-##timer 
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-client'  -f /data/nyc_taxi.csv
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca data spark_pandas failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#11 Total time cost ${time} seconds"
+#!/bin/bash
+echo "#11 start test for orca data spark_pandas"
+#timer 
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-client'  -f /data/nyc_taxi.csv
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca data spark_pandas failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#11 Total time cost ${time} seconds"
 
-echo "#12 start test for pytorch cifar10"
+echo "#12-1 start test for pytorch cifar10 yarn-client mode"
 #timer  
 start=$(date "+%s")
 #run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-client' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
 exit_status=$?
 if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch cifar10 failed"
+  echo "orca pytorch cifar10 yarn-client mode failed"
   exit $exit_status
 fi
 now=$(date "+%s")
 time=$((now - start))
 echo "#12 Total time cost ${time} seconds"
 
-#echo "#13 start test for pytorch fashion_mnist"
-##timer 
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode 'yarn-client'   --epochs 1  --batch_size 256 --download False --data_dir /data/fashion_mnist
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch fashion_mnist failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#13 Total time cost ${time} seconds"
-#
-#echo "#14 start test for pytorch super_resolution"
-##timer 
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode 'yarn-client' --data_dir /data/dataset 
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch super_resolution failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#14 Total time cost ${time} seconds"
-#
-#echo "#15 start test for orca torchmodel mnist"
-##timer   
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode 'yarn-client' --dir /data/test_mnist --download False
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel mnist failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#15 Total time cost ${time} seconds"
-#
-#echo "#16 start test for orca torchmodel imagenet"
-##timer  
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-client
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel imagenet failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#16 Total time cost ${time} seconds"
-#
-#echo "#17 start test for orca torchmodel resnet_finetune"
-##timer
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-client
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel resnet_finetune failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#17 Total time cost ${time} seconds"
-#
-#########################  cluster
-#echo "##18 start test for orca data spark_pandas"
-##timer 
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-cluster'  -f /data/nyc_taxi.csv
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca data spark_pandas failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#18 Total time cost ${time} seconds"
-#
-#echo "#19 start test for pytorch cifar10"
-##timer  
-## failed jep error
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode yarn-cluster --epochs 1  --batch_size 4 --data_dir /data/cifar10_data --download False --executor_memory '4g' --driver_memory '4g'
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch cifar10 failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#19 Total time cost ${time} seconds"
-#
-#echo "#20 start test for pytorch fashion_mnist"
-##timer 
-##failed jep error
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode yarn-cluster   --epochs 1  --batch_size 8 --download False --data_dir /data/fashion_mnist
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch fashion_mnist failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#20 Total time cost ${time} seconds"
-#
-#echo "#21 start test for orca torchmodel mnist"
-##timer  
-##failed jep error
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode yarn-cluster --dir /data/test_mnist --download False
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel mnist failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#21 Total time cost ${time} seconds"
-#
-#echo "#22 start test for pytorch super_resolution"
-##timer 
-##failed jep error
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode yarn-cluster --data_dir /data/dataset 
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch super_resolution failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#22 Total time cost ${time} seconds"
-#
-#echo "#23 start test for orca torchmodel imagenet"
-##timer   
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-cluster
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel imagenet failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#23 Total time cost ${time} seconds"
-#
-#echo "#24 start test for orca torchmodel resnet_finetune"
-##timer 
-##failed jep error
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-cluster
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca torchmodel resnet_finetune failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#24 Total time cost ${time} seconds"
-#
+echo "#12-2 start test for pytorch cifar10 yarn-cluster mode"
+#timer  
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch cifar10 yarn-cluster mode failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#12 Total time cost ${time} seconds"
+
+echo "#13 start test for pytorch fashion_mnist"
+#timer 
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode 'yarn-client'   --epochs 1  --batch_size 256 --download False --data_dir /data/fashion_mnist
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch fashion_mnist failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#13 Total time cost ${time} seconds"
+
+echo "#14 start test for pytorch super_resolution"
+#timer 
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode 'yarn-client' --data_dir /data/dataset 
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch super_resolution failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#14 Total time cost ${time} seconds"
+
+echo "#15 start test for orca torchmodel mnist"
+#timer   
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode 'yarn-client' --dir /data/test_mnist --download False
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel mnist failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#15 Total time cost ${time} seconds"
+
+echo "#16 start test for orca torchmodel imagenet"
+#timer  
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-client
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel imagenet failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#16 Total time cost ${time} seconds"
+
+echo "#17 start test for orca torchmodel resnet_finetune"
+#timer
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-client
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel resnet_finetune failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#17 Total time cost ${time} seconds"
+
+########################  cluster
+echo "##18 start test for orca data spark_pandas"
+#timer 
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-cluster'  -f /data/nyc_taxi.csv
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca data spark_pandas failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#18 Total time cost ${time} seconds"
+
+echo "#19 start test for pytorch cifar10"
+#timer  
+# failed jep error
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode yarn-cluster --epochs 1  --batch_size 4 --data_dir /data/cifar10_data --download False --executor_memory '4g' --driver_memory '4g'
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch cifar10 failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#19 Total time cost ${time} seconds"
+
+echo "#20 start test for pytorch fashion_mnist"
+#timer 
+#failed jep error
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode yarn-cluster   --epochs 1  --batch_size 8 --download False --data_dir /data/fashion_mnist
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch fashion_mnist failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#20 Total time cost ${time} seconds"
+
+echo "#21 start test for orca torchmodel mnist"
+#timer  
+#failed jep error
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode yarn-cluster --dir /data/test_mnist --download False
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel mnist failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#21 Total time cost ${time} seconds"
+
+echo "#22 start test for pytorch super_resolution"
+#timer 
+#failed jep error
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode yarn-cluster --data_dir /data/dataset 
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca pytorch super_resolution failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#22 Total time cost ${time} seconds"
+
+echo "#23 start test for orca torchmodel imagenet"
+#timer   
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-cluster
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel imagenet failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#23 Total time cost ${time} seconds"
+
+echo "#24 start test for orca torchmodel resnet_finetune"
+#timer 
+#failed jep error
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-cluster
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  echo "orca torchmodel resnet_finetune failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#24 Total time cost ${time} seconds"
+

--- a/python/orca/example/run-example-tests-yarn-integration.sh
+++ b/python/orca/example/run-example-tests-yarn-integration.sh
@@ -14,20 +14,20 @@
 #time=$((now - start))
 #echo "#11 Total time cost ${time} seconds"
 
-#echo "#12 start test for pytorch cifar10"
-##timer  
-#start=$(date "+%s")
-##run the example
-#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
-#exit_status=$?
-#if [ $exit_status -ne 0 ]; then
-#  clear_up
-#  echo "orca pytorch cifar10 failed"
-#  exit $exit_status
-#fi
-#now=$(date "+%s")
-#time=$((now - start))
-#echo "#12 Total time cost ${time} seconds"
+echo "#12 start test for pytorch cifar10"
+#timer  
+start=$(date "+%s")
+#run the example
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+exit_status=$?
+if [ $exit_status -ne 0 ]; then
+  clear_up
+  echo "orca pytorch cifar10 failed"
+  exit $exit_status
+fi
+now=$(date "+%s")
+time=$((now - start))
+echo "#12 Total time cost ${time} seconds"
 
 #echo "#13 start test for pytorch fashion_mnist"
 ##timer 

--- a/python/orca/example/run-example-tests-yarn-integration.sh
+++ b/python/orca/example/run-example-tests-yarn-integration.sh
@@ -14,20 +14,20 @@
 #time=$((now - start))
 #echo "#11 Total time cost ${time} seconds"
 
-echo "#12 start test for pytorch cifar10"
-#timer  
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch cifar10 failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#12 Total time cost ${time} seconds"
+#echo "#12 start test for pytorch cifar10"
+##timer  
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch cifar10 failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#12 Total time cost ${time} seconds"
 
 #echo "#13 start test for pytorch fashion_mnist"
 ##timer 

--- a/python/orca/example/run-example-tests-yarn-integration.sh
+++ b/python/orca/example/run-example-tests-yarn-integration.sh
@@ -1,24 +1,24 @@
-#!/bin/bash
-echo "#11 start test for orca data spark_pandas"
-#timer 
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-client'  -f /data/nyc_taxi.csv
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca data spark_pandas failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#11 Total time cost ${time} seconds"
+##!/bin/bash
+#echo "#11 start test for orca data spark_pandas"
+##timer 
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-client'  -f /data/nyc_taxi.csv
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca data spark_pandas failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#11 Total time cost ${time} seconds"
 
 echo "#12 start test for pytorch cifar10"
 #timer  
 start=$(date "+%s")
 #run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-client' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
+python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode 'yarn-cluster' --epochs 1  --batch_size 256 --data_dir /data/cifar10_data --download False
 exit_status=$?
 if [ $exit_status -ne 0 ]; then
   clear_up
@@ -29,189 +29,189 @@ now=$(date "+%s")
 time=$((now - start))
 echo "#12 Total time cost ${time} seconds"
 
-echo "#13 start test for pytorch fashion_mnist"
-#timer 
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode 'yarn-client'   --epochs 1  --batch_size 256 --download False --data_dir /data/fashion_mnist
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch fashion_mnist failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#13 Total time cost ${time} seconds"
-
-echo "#14 start test for pytorch super_resolution"
-#timer 
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode 'yarn-client' --data_dir /data/dataset 
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch super_resolution failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#14 Total time cost ${time} seconds"
-
-echo "#15 start test for orca torchmodel mnist"
-#timer   
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode 'yarn-client' --dir /data/test_mnist --download False
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel mnist failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#15 Total time cost ${time} seconds"
-
-echo "#16 start test for orca torchmodel imagenet"
-#timer  
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-client
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel imagenet failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#16 Total time cost ${time} seconds"
-
-echo "#17 start test for orca torchmodel resnet_finetune"
-#timer
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-client
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel resnet_finetune failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#17 Total time cost ${time} seconds"
-
-########################  cluster
-echo "##18 start test for orca data spark_pandas"
-#timer 
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-cluster'  -f /data/nyc_taxi.csv
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca data spark_pandas failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#18 Total time cost ${time} seconds"
-
-echo "#19 start test for pytorch cifar10"
-#timer  
-# failed jep error
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode yarn-cluster --epochs 1  --batch_size 4 --data_dir /data/cifar10_data --download False --executor_memory '4g' --driver_memory '4g'
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch cifar10 failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#19 Total time cost ${time} seconds"
-
-echo "#20 start test for pytorch fashion_mnist"
-#timer 
-#failed jep error
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode yarn-cluster   --epochs 1  --batch_size 8 --download False --data_dir /data/fashion_mnist
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch fashion_mnist failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#20 Total time cost ${time} seconds"
-
-echo "#21 start test for orca torchmodel mnist"
-#timer  
-#failed jep error
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode yarn-cluster --dir /data/test_mnist --download False
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel mnist failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#21 Total time cost ${time} seconds"
-
-echo "#22 start test for pytorch super_resolution"
-#timer 
-#failed jep error
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode yarn-cluster --data_dir /data/dataset 
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca pytorch super_resolution failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#22 Total time cost ${time} seconds"
-
-echo "#23 start test for orca torchmodel imagenet"
-#timer   
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-cluster
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel imagenet failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#23 Total time cost ${time} seconds"
-
-echo "#24 start test for orca torchmodel resnet_finetune"
-#timer 
-#failed jep error
-start=$(date "+%s")
-#run the example
-python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-cluster
-exit_status=$?
-if [ $exit_status -ne 0 ]; then
-  clear_up
-  echo "orca torchmodel resnet_finetune failed"
-  exit $exit_status
-fi
-now=$(date "+%s")
-time=$((now - start))
-echo "#24 Total time cost ${time} seconds"
-
+#echo "#13 start test for pytorch fashion_mnist"
+##timer 
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode 'yarn-client'   --epochs 1  --batch_size 256 --download False --data_dir /data/fashion_mnist
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch fashion_mnist failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#13 Total time cost ${time} seconds"
+#
+#echo "#14 start test for pytorch super_resolution"
+##timer 
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode 'yarn-client' --data_dir /data/dataset 
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch super_resolution failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#14 Total time cost ${time} seconds"
+#
+#echo "#15 start test for orca torchmodel mnist"
+##timer   
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode 'yarn-client' --dir /data/test_mnist --download False
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel mnist failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#15 Total time cost ${time} seconds"
+#
+#echo "#16 start test for orca torchmodel imagenet"
+##timer  
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-client
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel imagenet failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#16 Total time cost ${time} seconds"
+#
+#echo "#17 start test for orca torchmodel resnet_finetune"
+##timer
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-client
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel resnet_finetune failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#17 Total time cost ${time} seconds"
+#
+#########################  cluster
+#echo "##18 start test for orca data spark_pandas"
+##timer 
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/data/spark_pandas.py --deploy-mode 'yarn-cluster'  -f /data/nyc_taxi.csv
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca data spark_pandas failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#18 Total time cost ${time} seconds"
+#
+#echo "#19 start test for pytorch cifar10"
+##timer  
+## failed jep error
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/cifar10/cifar10.py --cluster_mode yarn-cluster --epochs 1  --batch_size 4 --data_dir /data/cifar10_data --download False --executor_memory '4g' --driver_memory '4g'
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch cifar10 failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#19 Total time cost ${time} seconds"
+#
+#echo "#20 start test for pytorch fashion_mnist"
+##timer 
+##failed jep error
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/fashion_mnist/fashion_mnist.py --cluster_mode yarn-cluster   --epochs 1  --batch_size 8 --download False --data_dir /data/fashion_mnist
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch fashion_mnist failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#20 Total time cost ${time} seconds"
+#
+#echo "#21 start test for orca torchmodel mnist"
+##timer  
+##failed jep error
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/mnist/main.py  --deploy-mode yarn-cluster --dir /data/test_mnist --download False
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel mnist failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#21 Total time cost ${time} seconds"
+#
+#echo "#22 start test for pytorch super_resolution"
+##timer 
+##failed jep error
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/learn/pytorch/super_resolution/super_resolution.py  --cluster_mode yarn-cluster --data_dir /data/dataset 
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca pytorch super_resolution failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#22 Total time cost ${time} seconds"
+#
+#echo "#23 start test for orca torchmodel imagenet"
+##timer   
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/imagenet/main.py  /data/imagenet2012 --batch-size 8 --max_epochs 1 --deploy_mode yarn-cluster
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel imagenet failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#23 Total time cost ${time} seconds"
+#
+#echo "#24 start test for orca torchmodel resnet_finetune"
+##timer 
+##failed jep error
+#start=$(date "+%s")
+##run the example
+#python ${BIGDL_ROOT}/python/orca/example/torchmodel/train/resnet_finetune/resnet_finetune.py /data/dogs_cats/samples --deploy-mode yarn-cluster
+#exit_status=$?
+#if [ $exit_status -ne 0 ]; then
+#  clear_up
+#  echo "orca torchmodel resnet_finetune failed"
+#  exit $exit_status
+#fi
+#now=$(date "+%s")
+#time=$((now - start))
+#echo "#24 Total time cost ${time} seconds"
+#

